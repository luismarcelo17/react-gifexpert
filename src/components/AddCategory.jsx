import React, { useState } from "react";
import propTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setinputValue] = useState("");
    const onInputChange = (event) => {
        // console.log(event.target.value);
        setinputValue(event.target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;

        // setCategories(categories => [inputValue, ...categories])
        onNewCategory(inputValue.trim());
        setinputValue("");
    };

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    );
};

AddCategory.propTypes = {
    onNewCategory: propTypes.func.isRequired,
};
