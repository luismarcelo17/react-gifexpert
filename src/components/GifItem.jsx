import propTypes from "prop-types";

export const GifItem = ({ id, title, url }) => {
    // console.log(title, url, id);

    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
};

/* Tarea

1. a√±adir PropTypes
    a. title obligatorio
    b. url obligatorio
2. Evaluar el spanshot

*/

GifItem.propTypes = {
    id: propTypes.string,
    title: propTypes.string.isRequired,
    url: propTypes.string.isRequired,
};
