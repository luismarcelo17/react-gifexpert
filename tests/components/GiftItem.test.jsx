import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en <GiftItem />", () => {
    const title = "sitama";
    const url = "https://onepuchn.com/saitama.jpg";

    test("Debe hacer martch con el snapshot", () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test("debe de mostrar la imagen con el URL y el ALT indicado", () => {
        render(<GifItem title={title} url={url} />);
        // Escribe el screen en consola o el sujeto de pruebas
        // screen.debug();

        // expect(screen.getByRole("img").src).toBe(url);
        // expect(screen.getByRole("img").alt).toBe(title);

        const { src, alt } = screen.getByRole("img");
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test("debe de mostrar el titulo en el componentes", () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getAllByText(title)).toBeTruthy();
    });
});
