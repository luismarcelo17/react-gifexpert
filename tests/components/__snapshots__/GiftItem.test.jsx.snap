// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem /> Debe hacer martch con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="sitama"
      src="https://onepuchn.com/saitama.jpg"
    />
    <p>
      sitama
    </p>
  </div>
</div>
`;
